spec/queries/display_ads/filtered_ads_query_spec.rb

```diff
@@ -1,154 +1,125 @@
-      display_ad.update!(published: true, approved: true)
-      expect(described_class.call(display_ads: DisplayAd.all, area: display_ad.placement_area,
-                                  organization_id: nil, user_signed_in: false)).to eq(display_ad)
+  context "when ads are not approved or published" do
+    let!(:unapproved) { create_display_ad approved: false }
+    let!(:unpublished) { create_display_ad published: false }
+    let!(:display_ad) { create_display_ad }
+
+    it "does not display unapproved or unpublished ads" do
+      filtered = filter_ads
+      expect(filtered).not_to include(unapproved)
+      expect(filtered).not_to include(unpublished)
+      expect(filtered).to include(display_ad)
     end
   end
 
   context "when considering article_tags" do
-    it "will show the display ads that contain tags that match any of the article tags" do
-      display_ad = create(:display_ad, organization_id: organization.id,
-                                       placement_area: "post_comments",
-                                       published: true,
-                                       approved: true,
-                                       cached_tag_list: "linux, git, go")
-
-      create(:display_ad, organization_id: organization.id,
-                          placement_area: "post_comments",
-                          published: true,
-                          approved: true,
-                          cached_tag_list: "career")
-
-      article_tags = %w[linux productivity]
-      expect(described_class.call(display_ads: DisplayAd.all, area: "post_comments", user_signed_in: false,
-                                  organization_id: nil, article_tags: article_tags)).to eq(display_ad)
-    end
+    let!(:no_tags) { create_display_ad cached_tag_list: "" }
+    let!(:mismatched) { create_display_ad cached_tag_list: "career" }
 
-    it "will show display ads that have no tags set" do
-      display_ad = create(:display_ad, organization_id: organization.id,
-                                       placement_area: "post_comments",
-                                       published: true,
-                                       approved: true,
-                                       cached_tag_list: "")
-
-      create(:display_ad, organization_id: organization.id,
-                          placement_area: "post_comments",
-                          published: true,
-                          approved: true,
-                          cached_tag_list: "career")
-
-      article_tags = %w[productivity java]
-      expect(described_class.call(display_ads: DisplayAd.all, area: "post_comments", user_signed_in: false,
-                                organization_id: display_ad.organization.id, article_tags: article_tags)).to eq(display_ad)
+    # This original test name seems misleading, it does not do what it says
+    pending "will show no display ads if the available display ads have no tags set or do not contain matching tags" do
```

Comment:
`cc` @Ridhwana @lboogie04 It can be hard to follow in the diff here, but in the [original version of these tests](https://github.com/forem/forem/blob/cfad413ca131c1d0b7a2954d50779f04be335700/spec/queries/display_ads/filtered_ads_query_spec.rb#L47-L74) there are two tests with sort of contradictory set ups and the name of this one is — ¿I _think_? — a _bit_ misleading. 

As I understand the filter behavior, when there _are_ tags on the article, we would show ads that _match_ the tags or any ads that _have **no** tags_ (as ads without tags are intended to match _all_ articles, not intended to _only_ match tag-less articles — I _think_?) The original version of this test does pass with `nil` filtered ads, but I _think_ that's because the setup has (in effect) _only_ created one ad (with a mis-matched tag list).

I included this test (as `pending`) since I wanted to confirm that I understood any of this right — because it's also possible we've got a bug here somehow?